infrastructure/code/code_html.go

```diff
@@ -30,172 +31,157 @@ import (
-	dataFlowHtml := `<table class="data-flow-body"><tbody>`
-
-	for i, flow := range issue.DataFlow {
-		fileName := filepath.Base(flow.FilePath)
-		dataFlowHtml += fmt.Sprintf(`
-		  <tr class="data-flow-row">
-		    <td class="data-flow-number">%d</td>
-		    <td class="data-flow-clickable-row" file-path="%s" start-line="%d" end-line="%d" start-character="%d" end-character="%d">%s:%d</td>
-		    <td class="data-flow-delimiter">|</td>
-		    <td class="data-flow-text">%s</td>
-		  </tr>`,
-			i+1,
-			html.EscapeString(flow.FilePath),
-			flow.FlowRange.Start.Line,
-			flow.FlowRange.End.Line,
-			flow.FlowRange.Start.Character,
-			flow.FlowRange.End.Character,
-			html.EscapeString(fileName),
-			flow.FlowRange.Start.Line+1,
-			html.EscapeString(flow.Content))
-	}
+func idxMinusOne(n int) int {
+	return n - 1
+}
 
-	dataFlowHtml += `</tbody></table>`
-	return dataFlowHtml
+func trimCWEPrefix(cwe string) string {
+	return strings.TrimPrefix(cwe, "CWE-")
 }
 
-func getCodeDiffHtml(fix snyk.ExampleCommitFix) string {
-	linesHtml := ""
-	for _, commit := range fix.Lines {
-		linesHtml += fmt.Sprintf(`
-		<div class="example-line %s"><code>%s</code></div>`, commit.LineChange, commit.Line)
+func getVisibilityClass(isIgnored bool) string {
+	if isIgnored {
+		return ""
 	}
-	return linesHtml
+	return "hidden"
 }
 
-func getTabsHtml(fixes []snyk.ExampleCommitFix) string {
-	tabsHtml := `<div class="tabs-nav">`
-
-	for i, fix := range fixes {
-		isSelectedClass := ""
-		if i == 0 {
-			isSelectedClass = "is-selected"
-		}
-
-		vendorIcon := getVendorIconSvg()
-		repoName := getRepoName(fix.CommitURL)
-		tabsHtml += fmt.Sprintf(`<span class="tab-item %s" id="tab-link-%d">%s %s</span>`, isSelectedClass, i, vendorIcon, repoName)
+func prepareIgnoreDetailsRow(ignoreDetails *snyk.IgnoreDetails) []IgnoreDetail {
+	return []IgnoreDetail{
+		{"Category", ignoreDetails.Category},
+		{"Ignored On", formatDate(ignoreDetails.IgnoredOn)},
+		{"Expiration", ignoreDetails.Expiration},
+		{"Ignored By", ignoreDetails.IgnoredBy},
 	}
+}
 
-	tabsHtml += `</div><div class="tab-container">`
+func prepareDataFlowTable(issue snyk.CodeIssueData) []DataFlowItem {
+	items := make([]DataFlowItem, 0, len(issue.DataFlow))
 
-	// Generate the contents for each tab
-	for i, fix := range fixes {
-		isSelectedClass := ""
-		if i == 0 {
-			isSelectedClass = "is-selected"
-		}
-		contentHtml := getCodeDiffHtml(fix)
-		tabsHtml += fmt.Sprintf(`<div id="tab-content-%d" class="tab-content %s">%s</div>`, i, isSelectedClass, contentHtml)
+	for i, flow := range issue.DataFlow {
+		items = append(items, DataFlowItem{
+			Number:         i + 1,
+			FilePath:       flow.FilePath,
+			StartLine:      flow.FlowRange.Start.Line,
+			EndLine:        flow.FlowRange.End.Line,
+			StartCharacter: flow.FlowRange.Start.Character,
+			EndCharacter:   flow.FlowRange.End.Character,
+			FileName:       filepath.Base(flow.FilePath),
+			Content:        flow.Content,
```

Comment:
You'll need a left trim here :)

Reply:
I have tried the template with the same issue reported by @Arvi3d and it seems that the rendering is taking care of the white space.

![no-trim](https://github.com/snyk/snyk-ls/assets/1948377/1df2ad5e-263d-49a3-bf0e-f87eec157dd6)